spylon/spark/yarn_launcher.py

```diff
@@ -317,14 +328,26 @@ def launcher(deploy_mode, args, working_dir="."):
         # Archive the conda environment
         env_archive = archive_dir(env_dir)
 
+        abs_env_dir = os.path.abspath(env_dir)
+        abs_env_archive = os.path.abspath(env_archive)
+
+        cleanup_functions.extend([
+            lambda: shutil.rmtree(abs_env_dir),
+            lambda: os.unlink(abs_env_archive),
+        ])
     else:
         raise NotImplementedError()
 
     args = dict(env_dir=env_dir, env_name=env_name, env_archive=env_archive, args=spark_args)
-    if deploy_mode == "client":
-        run_pyspark_yarn_client(**args)
-    elif deploy_mode == "cluster":
-        run_pyspark_yarn_cluster(**args)
+    try:
+        if deploy_mode == "client":
+            run_pyspark_yarn_client(**args)
+        elif deploy_mode == "cluster":
+            run_pyspark_yarn_cluster(**args)
+    finally:
+        if cleanup:
+            for fn in cleanup_functions:
```

Comment:
Maybe should try/except within this loop too in case any of the cleanup func blows up.